src/components/Button.res

```diff
@@ -698,107 +699,109 @@ let make = (
+            ? ""
+            : "justify-center"} ${relativeClass} ${heightClass} ${newThemeGap} ${conditionalButtonStyles} items-center ${borderStyle}  ${textColor} ${cursorType} ${paddingClass} ${lengthStyle} ${customButtonStyle}  ${customTextOverFlowClass}`}
+        onClick=handleClick>
+        {if buttonState == Loading {
+          <span className={iconPadding}>
+            <span className={`flex items-center mx-2 animate-spin`}>
+              <Loadericon size=iconSize iconColor=?loaderIconColor />
+            </span>
           </span>
-        </span>
-      } else {
-        switch leftIcon {
+        } else {
+          switch leftIcon {
+          | FontAwesome(iconName) =>
+            <span className={`flex items-center ${iconColor} ${iconMargin} ${iconPadding}`}>
+              <Icon
+                className={`align-middle ${strokeColor} ${iconBorderColor}`}
+                size=iconSize
+                name=iconName
+              />
+            </span>
+          | Euler(iconName) =>
+            <span className={`flex items-center ${iconColor} ${iconMargin} ${eulerIconPadding}`}>
+              <Icon className={`align-middle ${strokeColor}`} size=iconSize name=iconName />
+            </span>
+          | CustomIcon(element) =>
+            <span className={`flex items-center ${iconMargin}`}> {element} </span>
+          | _ => React.null
+          }
+        }}
+        {switch text {
+        | Some(textStr) =>
+          if textStr !== "" {
+            let btnContent =
+              <AddDataAttributes attributes=[("data-button-text", textStr)]>
+                <div
+                  className={`${textPaddingClass} ${textSize} ${textWeight} ${ellipsisClass} whitespace-pre ${textStyle}`}>
+                  {buttonState == Loading ? React.string(loadingText) : React.string(textStr)}
+                </div>
+              </AddDataAttributes>
+
+            if showBtnTextToolTip {
+              <div className=ellipsisParentClass>
+                <ToolTip
+                  description=textStr
+                  toolTipFor=btnContent
+                  contentAlign=Default
+                  justifyClass="justify-start"
+                />
+              </div>
+            } else {
+              <div className=ellipsisParentClass> btnContent </div>
+            }
+          } else {
+            React.null
+          }
+
+        | None => React.null
+        }}
+        {switch badge.color {
+        | NoBadge => React.null
+        | _ =>
+          <AddDataAttributes attributes=[("data-badge-value", badge.value)]>
+            <span
+              className={`flex items-center ${rightIconSpacing} ${badgeColor} ${badgeTextColor} ${badgeSpacing} ${badgeTextSize}  rounded-full`}>
+              {React.string(badge.value)}
+            </span>
+          </AddDataAttributes>
+        }}
+        {switch buttonRightText {
+        | Some(text) =>
+          <UIUtils.RenderIf condition={text !== ""}>
```

Comment:
use util function to check for empty string

Reply:
Moved this function to LogicUtils and used that function